engine/runtime/src/main/java/org/enso/compiler/TreeToIr.java

```diff
@@ -550,7 +550,14 @@ IR.Expression translateExpression(Tree tree, boolean isMethod) {
         yield switch (op.codeRepr()) {
           case "." -> {
             final Option<IdentifiedLocation> loc = getIdentifiedLocation(tree);
-            yield buildQualifiedName(app, loc, false);
+            try {
+              yield buildQualifiedName(app, loc, false);
+            } catch (UnhandledEntity ex) {
+              if (ex.entity() instanceof IR.Expression expr) {
```

Comment:
I'm not a fan of exceptions being part of a regular flow of the execution. This can quickly get out of hand.